packages/node-core/src/indexer/connectionPool.service.ts

```diff
@@ -185,19 +189,19 @@ export class ConnectionPoolService<T extends IApiConnectionSpecific<any, any, an
     return Object.keys(this.pool).length;
   }
 
-  async handleApiDisconnects(apiIndex: number, endpoint: string): Promise<void> {
-    logger.warn(`disconnected from ${endpoint}`);
-    delete this.pool[apiIndex];
+  async handleApiDisconnects(apiIndex: number): Promise<void> {
+    logger.warn(`disconnected from ${this.pool[apiIndex].endpoint}`);
 
     try {
-      logger.debug(`reconnecting to ${endpoint}...`);
+      logger.debug(`reconnecting to ${this.pool[apiIndex].endpoint}...`);
       await this.connectToApi(apiIndex);
     } catch (e) {
-      logger.error(`unable to reconnect to endpoint ${endpoint}`, e);
+      logger.error(`unable to reconnect to endpoint ${this.pool[apiIndex].endpoint}: ${e}`);
+      delete this.pool[apiIndex];
```

Comment:
So we only retry connecting once? Why not try multiple times with exponential backoff.